### PR TITLE
Add support for text tracks

### LINKED ISSUE #36: Subtitles support
References:

- https://www.extremraym.com/en/wavesurfer-wp-markers/
- http://docs.videojs.com/tutorial-text-tracks.html

### PR DESCRIPTION
**Pull request by mfairchild365**
This creates a 'player technology' (tech) that extends the default HTML5 tech and then replaces the HTML5 tech. By extending the HTML5 tech, we can have it properly emit events that trigger text track functionality in VideoJS.

closes #36

### CODE REVIEW

**Comment by thijstriemstra** [comment]
Wow, this is huge, thanks, I will take a look!

**Comment by mfairchild365** [comment]
You are welcome. Thank you for merging!

**Comment by thijstriemstra** [comment]
@mfairchild365 I read your extensive commit messages, and you're basically saying most of the logic should move to a tech? I'm not so familiar with those (and this plugin started out as a 'hack' to re-use the video.js controls with wavesurfer.js tech) but if that's the recommended way to implement stuff we should look at that.

**Comment by thijstriemstra** [comment]
and is it possible to enable the captions at startup? the example now starts before you have time to click CC btn.

**Comment by mfairchild365** [comment]
Yep, thats how I understand it. The 'tech' should handle the playback, while the plugin should change the interface/controls. I think it would be best to move most of the functionality into the tech rather than the plugin. Unfortunately, I didn't have time do that much work, so I made it work with a minimal number of changes. Perhaps it could be revisited in the future. Full disclosure: I'm honestly not very familiar with videoJS (this is my first time working with it).

To auto-show the captions, you can add the `default` attribute to the caption track that you want to display. 

```
<track kind="captions" src="sample.vtt" srclang="en-US" default />
```

File: src/js/videojs.wavesurfer.js [hunk 1]
```diff
@@ -14,10 +14,11 @@ import videojs from 'video.js';
 import WaveSurfer from 'wavesurfer.js';
 
 const Plugin = videojs.getPlugin('plugin');
+const Tech = videojs.getComponent('Tech');
```
Comment (thijstriemstra) [inline src/js/videojs.wavesurfer.js]: @mfairchild365 I cannot find any references to this const so it's safe to remove this Tech import right?

Comment (mfairchild365) [reply src/js/videojs.wavesurfer.js]: Yes. I actually meant to remove that. The new tech was originally extending `Tech`, but I changed it to extend the 'HTML5' tech. Good catch.

Comment (thijstriemstra) [reply src/js/videojs.wavesurfer.js]: cool. cleaned it up a little and moved it to a separate module. Will release as 2.1.0, probably today. Cheers.

**Comment by thijstriemstra** [comment]
When this plugin was created (around video.js 4.x) audio was not supported by video.js so I took this approach. Now that video.js does support audio this plugin could be refactored a little. As long as we prevent that the audio file has to be loaded twice, once for video.js and once for wavesurfer.js.

**Pull request merged**